hal/mbed_sleep_manager.c

```diff
@@ -20,13 +20,62 @@
 #include "sleep_api.h"
 #include "mbed_error.h"
 #include "mbed_debug.h"
+#include "mbed_stats.h"
+#include "lp_ticker_api.h"
 #include <limits.h>
 #include <stdio.h>
+#include "mbed_stats.h"
+
 
 #if DEVICE_SLEEP
 
 // deep sleep locking counter. A target is allowed to deep sleep if counter == 0
 static uint16_t deep_sleep_lock = 0U;
+static uint64_t sleep_time = 0;
+static uint64_t deep_sleep_time = 0;
+
+#if defined(MBED_CPU_STATS_ENABLED) && defined(DEVICE_LOWPOWERTIMER)
+static ticker_data_t *sleep_ticker = NULL;
+#endif
+
+static inline uint64_t read_us(void)
+{
+#if defined(MBED_CPU_STATS_ENABLED) && defined(DEVICE_LOWPOWERTIMER)
+    if (NULL == sleep_ticker) {
+        sleep_ticker = (ticker_data_t*) get_lp_ticker_data();
+    }
+    return ticker_read_us(sleep_ticker);
+#else
+    return 0;
+#endif
+}
+
+uint64_t mbed_time_idle(void)
+{
+    return (sleep_time+deep_sleep_time);
+}
+
+uint64_t mbed_uptime(void)
```

Comment:
`mbed_uptime` is the same as `read_us` can we remove one? or at least make one simply call the other?